services/core-web/src/components/Forms/noticeOfWork/UpdateNOWDateForm.js

```diff
@@ -0,0 +1,300 @@
+import React from "react";
+import { Field, reduxForm, getFormValues } from "redux-form";
+import { Button, Popconfirm, Row, Col, Alert } from "antd";
+import { compose } from "redux";
+import { connect } from "react-redux";
+import { isEmpty } from "lodash";
+import { Form } from "@ant-design/compatible";
+import "@ant-design/compatible/assets/index.css";
+import { resetForm, formatDate } from "@common/utils/helpers";
+import PropTypes from "prop-types";
+import { required, dateNotInFuture, date } from "@common/utils/Validate";
+import * as FORM from "@/constants/forms";
+import { renderConfig } from "@/components/common/config";
+import AuthorizationWrapper from "@/components/common/wrappers/AuthorizationWrapper";
+import * as Permission from "@/constants/permissions";
+
+const propTypes = {
+  title: PropTypes.bool.isRequired,
+  handleSubmit: PropTypes.func.isRequired,
+  initialValues: PropTypes.objectOf(PropTypes.any).isRequired,
+  showCommentFields: PropTypes.string.isRequired,
+  importedDate: PropTypes.string.isRequired,
+  closeModal: PropTypes.func.isRequired,
+  recordType: PropTypes.string.isRequired,
+};
+
+const recordTypeCodes = {
+  delay: "DEL",
+  progress: "PRO",
+  verification: "VER",
+  decision: "DEC",
+};
+
+const getPreviousDelayStartDate = (rowIndex, values) => {
+  const length = values.delays.length;
+  const isCurrentLast = rowIndex + 1 === length;
+  const isCurrentFirst = rowIndex === 0;
+  let dates = {};
+  if (length > 0) {
+    if (isCurrentFirst && isCurrentLast) {
+      dates = {};
+    } else if (isCurrentFirst) {
+      dates = {
+        prev_end_date: values.delays[values?.rowIndex + 1].end_date,
+        next_start_date: null,
+      };
+    } else if (isCurrentLast) {
+      dates = {
+        prev_end_date: null,
+        next_start_date: values.delays[values?.rowIndex - 1].start_date,
+      };
+    } else {
+      dates = {
+        prev_end_date: values.delays[values?.rowIndex + 1].end_date,
+        next_start_date: values.delays[values?.rowIndex - 1].start_date,
+      };
+    }
+  }
+  return dates;
+};
+
+const getDateWithoutTime = (date) => date && date.substring(0, 10);
```

Comment:
I am a bit worried about substring on a date. will this always work or could there be a format of date where this will break?

Reply:
As far as I can tell, this will always work. It just removes all characters after 10 - so all dates will look like "2020-09-01"